wizard/archive_purchase_order.py

```diff
@@ -0,0 +1,23 @@
+from tokenize import group
+from odoo import api, models, fields
+from odoo.exceptions import UserError
+
+
+class ArchivePurchaseOrder(models.TransientModel):
+    _name = 'archive.purchase.order'
+    _description = 'Archive Multiple PO'
+
+    archive_purchase_order_ids = fields.Many2many(
+        'purchase.order',
+        string="Purchase Orders"
+    )
+    button_archive_invisible = fields.Boolean(
+        compute="_compute_button_invisible")
+
+    @api.depends('archive_purchase_order_ids')
+    def _compute_button_invisible(self):
+        self.button_archive_invisible = False if len(
+            self.archive_purchase_order_ids) > 0 else True
+
+    def action_archive(self):
+        self.archive_purchase_order_ids._check_accessability()
```

Comment:
It's the action for archiving orders, but you only checked accessibility???

Reply:
I really appologize. It should be `self.archive_purchase_order_ids.action_archive()`